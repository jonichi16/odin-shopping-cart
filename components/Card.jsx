import React from 'react';
import Image from 'next/image';
import AddToCartButton from './AddToCartButton';

const Card = ({ product }) => {
  return (
    <div className='bg-slate-50 w-fit rounded-md shadow'>
      <Image
        priority
        src={product.image.src}
        width={250}
        height={250}
        alt={product.name}
        className='p-3'
      />
      <div className='border-t-2 px-3 py-5'>
        <h2 className='text-lg font-bold pb-2'>{product.name}</h2>
        <div className='flex items-center'>
          <p className='text-red-500 text-sm font-bold flex-1'>
            {product.price} Gil
          </p>
          <AddToCartButton product={product} />
        </div>
      </div>
    </div>
  );
};

export default Card;
