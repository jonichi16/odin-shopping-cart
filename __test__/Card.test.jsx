import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/user-event';
import Card from '../components/Card';

describe('Card component', () => {
  const product = {
    id: 1,
    image: {
      src: '/test.png',
      width: 250,
      height: 250,
    },
    name: 'Sample Item',
    price: 1200,
  };

  it('renders the correct component', () => {
    const { container } = render(<Card product={product} />);

    expect(container).toMatchSnapshot();
  });
});
