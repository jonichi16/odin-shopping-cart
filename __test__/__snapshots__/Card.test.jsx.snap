// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the correct component 1`] = `
<div>
  <div
    class="bg-slate-50 w-fit rounded-md shadow"
  >
    <img
      alt="Sample Item"
      class="p-3"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="250"
      src="/_next/image?url=%2Ftest.png&w=640&q=75"
      srcset="/_next/image?url=%2Ftest.png&w=256&q=75 1x, /_next/image?url=%2Ftest.png&w=640&q=75 2x"
      style="color: transparent;"
      width="250"
    />
    <div
      class="border-t-2 px-3 py-5"
    >
      <h2
        class="text-lg font-bold pb-2"
      >
        Sample Item
      </h2>
      <div
        class="flex items-center"
      >
        <p
          class="text-red-500 text-sm font-bold flex-1"
        >
          1200
           Gil
        </p>
        <button
          class="bg-blue-800 text-slate-50 py-1 px-5 uppercase font-bold text-sm rounded-full hover:opacity-90 active:scale-95"
        >
          Add to Cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
